v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java

```diff
@@ -272,13 +273,26 @@ private void listTables(Ddl.Builder builder) {
       // Parent table and child table has to be in same schema.
       String parentTableName =
           resultSet.isNull(2) ? null : getQualifiedName(tableSchema, resultSet.getString(2));
-      String onDeleteAction = resultSet.isNull(3) ? null : resultSet.getString(3);
+      String interleaveTypeStr = resultSet.isNull(3) ? null : resultSet.getString(3);
+      Table.InterleaveType interleaveType = null;
+      if (!Strings.isNullOrEmpty(interleaveTypeStr)) {
+        interleaveType =
+            interleaveTypeStr.equals("IN PARENT") ? InterleaveType.IN_PARENT : InterleaveType.IN;
+      }
+      String onDeleteAction = resultSet.isNull(4) ? null : resultSet.getString(4);
 
       // Error out when the parent table or on delete action are set incorrectly.
-      if (Strings.isNullOrEmpty(parentTableName) != Strings.isNullOrEmpty(onDeleteAction)) {
+      if (Strings.isNullOrEmpty(parentTableName) != Strings.isNullOrEmpty(interleaveTypeStr)) {
+        throw new IllegalStateException(
+            String.format(
+                "Invalid combination of parentTableName %s and interleaveType %s",
+                parentTableName, interleaveType));
+      }
+
+      if ((interleaveType == InterleaveType.IN_PARENT) == Strings.isNullOrEmpty(onDeleteAction)) {
```

Comment:
Why is this an invalid combination? If interleaveType is IN PARENT, onDeleteAction cannot be null ?

Reply:
Actually the if condition expressions in both the above if blocks are super confusing. Either add comments explaining scenarios or avoid using == and != to compare boolean operands and expand the expr with && ||

Reply:
"If interleaveType is IN PARENT, onDeleteAction cannot be null ?" -> yes that is correct.

Sure thing though, I'll add comments